modules/twinklebatchprotect.js

```diff
@@ -281,9 +281,8 @@ Twinkle.batchprotect.callback = function twinklebatchprotectCallback() {
 		query.gcmlimit = Twinkle.getPref('batchMax'); // the max for sysops
 	} else if (mw.config.get('wgCanonicalSpecialPageName') === 'Prefixindex') {
 		query.generator = 'allpages';
-		query.gapnamespace = Morebits.queryString.exists('namespace') ? Morebits.queryString.get('namespace') : $('select[name=namespace]').val();
-		query.gapprefix = Morebits.queryString.exists('from') ? Morebits.string.toUpperCaseFirstChar(Morebits.queryString.get('from').replace('+', ' ')) :
-			Morebits.string.toUpperCaseFirstChar($('input[name=prefix]').val());
+		query.gapnamespace = mw.util.getParamValue('namespace') || $('select[name=namespace]').val();
+		query.gapprefix = mw.util.getParamValue('prefix') || $('input[name=prefix]').val();
```

Comment:
Wow!  `from` is used to paginate, while `prefix` is the actual prefix.  That is, `prefix` defines the list (e.g. [`prefix=A`](https://en.wikipedia.org/wiki/Special:PrefixIndex?prefix=A&namespace=0)) while `from` defines what is shown on the user's screen (e.g. [`prefix=A&from=Amory_Lovins`](https://en.wikipedia.org/wiki/Special:PrefixIndex?prefix=A&from=Amory_Lovins&namespace=0)).  `batchprotect` has been busted on non-first-page results for years!  Very lame!